new_dawn_server/users/api/resources.py

```diff
@@ -239,6 +243,19 @@ def _get_model_fields_dict(bundle, fields):
     def _get_account_name(first_name, last_name):
         return first_name + ACCOUNT_NAME_DELIMITER + last_name
 
+    @staticmethod
+    def deserialize_city_pref(bundle, account):
```

Comment:
Rename this function to get_and_save_city_pref, and makes this function only saves city_pref and return city_pref, removing the logic for appending to account.

Notice that the terminology "deserialize" corresponds to unwrap serialized thing, here you don't have anything truelly serialized.